### PR TITLE
constructing post endpoint first

### PR DESCRIPTION
**Pull request by motola**
What does this PR do?
Create POST and GET requests to the route /posts

Description of Task to be completed?
setup routes, controllers and models, mongoose ODM and minor validation (which would be updated to a middleware later on)

How should this be manually tested?
clone the repo
run npm install

What are the relevant Trello User stories?
https://trello.com/c/ktDvvFbE

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "stackoverflow-api",
+  "version": "1.0.0",
+  "description": "```sh\r  A user should create an account and login\r ```\r ```sh\r A user should be able to create a post.\r ```\r ```sh\r A user should be able to change his or her post.\r ```\r ```sh\r A user should be able to delete a post.\r ```\r ```sh\r A user should be able to view the created post\r ```\r ```sh\r A user should be able to view all the created posts.\r ```\r ```sh\r A user should be able to search for a post\r ```",
```
Comment (ludralph) [inline package.json]: The description should describe what the app does and not the user stories as you presented here.
for example:  An app to manage blog posts

File: router/postsRoutes.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const express = require('express');
+const router = express.Router();
+const { getPosts, setPosts } = require('../controllers/postsController');
+
+
+
+
+router.get('/', getPosts);
+router.post('/', setPosts)
```
Comment (ludralph) [inline router/postsRoutes.js]: what does setPosts mean? Always strive to use meaningful names. for example `createPosts` simply tells the user that we want to create a new post.

File: model/postModel.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const mongoose = require('mongoose');
+const nodemon = require('nodemon');
+
+// const tagSchema = new mongoose.Schema({
+//     name: String,
+//     });
+const commentSchema = new mongoose.Schema({
```
Comment (ludralph) [inline model/postModel.js]: Are you using `commentSchema`? If it's not used, kindly remove it.

File: controllers/postsController.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+const Post = require('../model/postModel')
+
+const getPosts = async (req, res) => {
+    const posts = await Post.find();
+
+    res.status(200).json(posts); 
+};
+
+
+const setPosts = async (req, res) => {
```
Comment (ludralph) [inline controllers/postsController.js]: please rename `setPosts` to `createPosts`

**Review by ludralph — CHANGES_REQUESTED** [review]

**Review by motola — COMMENTED** [review]
Changes have been made. Thank you for the feedback

**Review by motola — COMMENTED** [review]
Additional changes made

**Review by ludralph — APPROVED** [review]

**Pull request merged**